pkg/provider/gitlab/gitlab.go

```diff
@@ -285,14 +335,23 @@ func (g *Gitlab) Close(ctx context.Context) error {
 	return nil
 }
 
-// Validate will use the gitlab client to validate the gitlab provider using the ListVariable call to ensure get permissions without needing a specific key.
+// Validate will use the gitlab projectClient to validate the gitlab provider using the ListVariable call to ensure get permissions without needing a specific key.
 func (g *Gitlab) Validate() (esv1beta1.ValidationResult, error) {
-	_, resp, err := g.client.ListVariables(g.projectID, nil)
+	_, resp, err := g.projectClient.ListVariables(g.projectID, nil)
 	if err != nil {
 		return esv1beta1.ValidationResultError, fmt.Errorf(errList, err)
 	} else if resp == nil || resp.StatusCode != http.StatusOK {
-		return esv1beta1.ValidationResultError, fmt.Errorf(errAuth)
+		return esv1beta1.ValidationResultError, fmt.Errorf(errProjectAuth, g.projectID)
 	}
+	for _, groupID := range g.groupIDs {
+		_, resp, err := g.groupClient.ListVariables(groupID, nil)
+		if err != nil {
+			return esv1beta1.ValidationResultError, fmt.Errorf(errList, err)
+		} else if resp == nil || resp.StatusCode != http.StatusOK {
+			return esv1beta1.ValidationResultError, fmt.Errorf(errGroupAuth, groupID)
+		}
+	}
+
```

Comment:
so, who exactly do you merge variables having same name/key from "group" and "project" ?

Reply:
The logic is around lines 198 (GetAllSecrets) and 257 (GetSecret).

Reply:
thx for the info @domizei385